baseapp/baseapp.go

```diff
@@ -392,23 +406,29 @@ func handleQueryP2P(app *BaseApp, path []string, req abci.RequestQuery) (res abc
 	return sdk.ErrUnknownRequest(msg).QueryResult()
 }
 
-// Implements ABCI
+// BeginBlock implements the ABCI application interface.
 func (app *BaseApp) BeginBlock(req abci.RequestBeginBlock) (res abci.ResponseBeginBlock) {
-	// Initialize the DeliverTx state.
-	// If this is the first block, it should already
-	// be initialized in InitChain.
-	// Otherwise app.deliverState will be nil, since it
-	// is reset on Commit.
+	if app.cms.TracingEnabled() {
+		app.cms.WithTracingContext(&sdk.TraceContext{
```

Comment:
To add transaction hashes, add `TxHash` to `sdk.TraceContext` and set it in `runTx`.